src/backend/looped/vehicle/vehicle_ammo_special_type.cpp

```diff
@@ -0,0 +1,248 @@
+			explosion.m_hit_plane                              = explosion_tag;
+			explosion.m_hit_truck                              = explosion_tag;
+			m_mg_weapon_info->m_explosion                      = explosion;
+			m_mg_weapon_info->m_ammo_info->m_ammo_special_type = ammo_type;
+			m_mg_weapon_info->m_speed                          = g_vehicle_ammo_settings.speed;
+			m_mg_weapon_info->m_time_between_shots             = g_vehicle_ammo_settings.time_between_shots;
+			m_mg_weapon_info->m_alternate_wait_time            = g_vehicle_ammo_settings.alternate_wait_time;
+			m_mg_weapon_info->m_weapon_range                   = g_vehicle_ammo_settings.weapon_range;
+		}
+
+		void restore_rocket()
+		{
+			if (m_rocket_weapon_info != nullptr)
+			{
+				m_rocket_weapon_info->m_damage_type         = m_rocket_damage_type;
+				m_rocket_weapon_info->m_time_between_shots  = m_rocket_time_between_shots;
+				m_rocket_weapon_info->m_alternate_wait_time = m_rocket_alternate_wait_time;
+				m_rocket_weapon_info->m_lock_on_range       = m_rocket_lock_on_range;
+				m_rocket_weapon_info->m_weapon_range        = m_rocket_range;
+				m_rocket_weapon_info->m_reload_time_mp      = m_rocket_reload_time_mp;
+				m_rocket_weapon_info->m_reload_time_sp      = m_rocket_reload_time_sp;
+				weapon_flags(m_rocket_weapon_info)          = m_rocket_weapon_flags;
+
+				CAmmoRocketInfo* rocket_info        = (CAmmoRocketInfo*)m_rocket_weapon_info->m_ammo_info;
+				rocket_info->m_explosion            = m_rocket_explosion;
+				rocket_info->m_lifetime             = m_rocket_lifetime;
+				rocket_info->m_launch_speed         = m_rocket_launch_speed;
+				rocket_info->m_time_before_homing   = m_rocket_time_before_homing;
+				rocket_info->m_homing_rocket_params = m_rocket_homing_params;
+				m_rocket_weapon_info                = nullptr;
+			}
+		}
+
+		void backup_rocket(CWeaponInfo* weapon_info)
+		{
+			m_rocket_weapon_info         = weapon_info;
+			m_rocket_time_between_shots  = weapon_info->m_time_between_shots;
+			m_rocket_alternate_wait_time = weapon_info->m_alternate_wait_time;
+			m_rocket_lock_on_range       = weapon_info->m_lock_on_range;
+			m_rocket_range               = weapon_info->m_weapon_range;
+			m_rocket_reload_time_mp      = weapon_info->m_reload_time_mp;
+			m_rocket_reload_time_sp      = weapon_info->m_reload_time_sp;
+			m_rocket_weapon_flags        = weapon_flags(weapon_info);
+
+			CAmmoRocketInfo* rocket_info = (CAmmoRocketInfo*)weapon_info->m_ammo_info;
+			m_rocket_explosion           = rocket_info->m_explosion;
+			m_rocket_lifetime            = rocket_info->m_lifetime;
+			m_rocket_launch_speed        = rocket_info->m_launch_speed;
+			m_rocket_time_before_homing  = rocket_info->m_time_before_homing;
+			m_rocket_homing_params       = rocket_info->m_homing_rocket_params;
+		}
+
+		void apply_rocket(const vehicle_ammo_setting& g_vehicle_ammo_settings)
+		{
+			m_rocket_weapon_info->m_damage_type         = eDamageType::Explosive;
```

Comment:
Why are you checking if m_rocket_weapon_info is not a null pointer before running the code in restore_rocket but not in apply_rocket? Is it impossible for m_rocket_weapon_info to be in a dirty state for apply_rocket?